spec/controllers/grades_controller_spec.rb

```diff
@@ -41,20 +42,42 @@
         expect(controller.instance_variable_get(:@questions)[:review1].size).to eq(1)
         expect(response).to render_template(:view)
       end
-    end
+      end
 
-    context 'when current assignment does not vary rubric by round' do
-      it 'calculates scores and renders grades#view page' do
-        allow(AssignmentQuestionnaire).to receive(:where).with(assignment_id: 1, used_in_round: 2).and_return([])
-        allow(ReviewResponseMap).to receive(:get_assessments_for).with(team).and_return([review_response])
-        params = {id: 1}
-        get :view, params
-        expect(controller.instance_variable_get(:@questions)[:review].size).to eq(1)
-        expect(response).to render_template(:view)
+    # This test is expected to fail due to the commented code on lines 38-42 in grades_controller. One of the aims of
+    # E2111
+    # is to decouple this code from the grades controller, where both the code itself and the testing are
+    # significantly coupled into the grades controller.
+    xcontext 'when user hasn\'t logged in to GitHub' do
```

Comment:
Any reasons why you comment out these test scenarios?

Reply:
We are moving the github metrics functionality out of the grades controller and we were looking at tests which were failing because of commented out github code in grades_controller. This particular test is being moved over to the new metrics_controller and will be re-enabled to cover authentication once it's moved to the new spec file. We expect the updated test code to be pushed tonight or tomorrow morning.

Reply:
I needed to resolve a merge conflict on this file so went ahead and reverted the xcontext at the same time.